util/internal/common/src/main/java/org/hibernate/search/v6poc/util/impl/common/SuppressingCloser.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.util.impl.common;
+
+/**
+ * A helper for closing multiple resources and re-throwing a provided exception,
+ * {@link Throwable#addSuppressed(Throwable) suppressing} any exceptions caught while closing.
+ * <p>
+ * This class is not thread safe.
+ * <p>
+ * This helper is mainly useful when implementing a {@code catch} block where resources must be closed,
+ * to make sure that all resources are at least given the chance to close, even if closing one of them fails,
+ * and that you can still re-throw the originally caught exception.
+ * <p>
+ * See the {@link AbstractCloser} superclass for a list of methods
+ * allowing to close objects while catching exceptions.
+ *
+ * @author Yoann Rodiere
+ */
+public final class SuppressingCloser extends AbstractCloser<SuppressingCloser, Exception> {
```

Comment:
Not sure about the name as it does not really suppress anything. It's more about merging the exceptions with the main one. That doesn't make a good name though :).